src/helpers/setProp.js

```diff
@@ -0,0 +1,36 @@
+/** @license ISC License (c) copyright 2018 original and current authors */
+/** @author Ian Hofmann-Hicks (evil) */
+
+const curry = require('../core/curry')
+const isObject = require('../core/isObject')
+const isString = require('../core/isString')
+const object = require('../core/object')
+
+function fn(name) {
+  function setProp(key, val, x) {
+    if(isString(key)) {
+      if(isObject(x)) {
+        return object.set(key, val , x)
+      }
+
+      throw new TypeError(
+        `${name}: Object required for third argument`
```

Comment:
I see that `setPath` still allows arrays and coerces them into objects. Do you think that it should throw for Objects too, to keep behaviour consistent?

Reply:
Good 👁!
Are you proposing that if the end of the path is an `Array` we throw?

Reply:
Yeah pretty much, for consistency's sake.